core/src/main/java/hudson/model/View.java

```diff
@@ -1317,14 +1319,17 @@ public void updateByXml(Source source) throws IOException {
     public static final Permission CONFIGURE = new Permission(PERMISSIONS,"Configure", Messages._View_ConfigurePermission_Description(), Permission.CONFIGURE, PermissionScope.ITEM_GROUP);
     public static final Permission READ = new Permission(PERMISSIONS,"Read", Messages._View_ReadPermission_Description(), Permission.READ, PermissionScope.ITEM_GROUP);
 
+    @SuppressFBWarnings(value = "RV_RETURN_VALUE_IGNORED_NO_SIDE_EFFECT", justification = "to guard against potential future compiler optimizations")
     @Initializer(before = InitMilestone.SYSTEM_CONFIG_LOADED)
     @Restricted(DoNotUse.class)
     public static void registerPermissions() {
         // Pending JENKINS-17200, ensure that the above permissions have been registered prior to
         // allowing plugins to adapt the system configuration, which may depend on these permissions
         // having been registered. Since this method is static and since it follows the above
         // construction of static permission objects (and therefore their calls to
-        // PermissionGroup#register), there is nothing further to do in this method.
+        // PermissionGroup#register), there is nothing further to do in this method. We call
+        // Objects.hash() to guard against potential future compiler optimizations.
+        Objects.hash(PERMISSIONS, CREATE, DELETE, CONFIGURE, READ);
```

Comment:
FTR this implements https://github.com/jenkinsci/jenkins/pull/5723/files#r707375258